frontend/public/module/k8s/network.ts

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { useK8sGet } from '../../components/utils/k8s-get-hook';
+import { K8sResourceKind } from './types';
+import { NetworkOperatorConfigModel } from '../../models';
+
+const clusterNetworkModel = NetworkOperatorConfigModel;
+const networkName = 'cluster';
+
+/**
+ * CNI type, or Unknown if the type can't be fetched (e.g. because the logged user does not
+ * have permissions to fetch it)
+ */
+enum ClusterNetworkType {
+  OpenShiftSDN = 'OpenShiftSDN',
+  OVNKubernetes = 'OVNKubernetes',
+  Unknown = 'Unknown',
+}
+
+export enum ClusterNetworkFeature {
+  PolicyEgress = 'PolicyEgress',
+  PolicyPeerIPBlockExceptions = 'PolicyPeerIPBlockExceptions',
+}
+
+export type ClusterNetworkFeatures = {
+  [key in ClusterNetworkFeature]?: boolean;
+};
+
+/**
+ * Main document depicting all the features that are supported by each supported CNI type.
+ * Undefined features would require to take an ambiguous action (e.g. allow the customer
+ * to set the policy Egress rule in a form, but show a warning explaining that this field is
+ * not available for the Openshift SDN type)
+ */
+const featuresDocument: {
+  [k in ClusterNetworkType]: ClusterNetworkFeatures;
+} = {
+  OpenShiftSDN: {
+    PolicyEgress: false,
+    PolicyPeerIPBlockExceptions: false,
+  },
+  OVNKubernetes: {
+    PolicyEgress: true,
+    PolicyPeerIPBlockExceptions: true,
+  },
+  Unknown: {},
+};
+
+/**
+ *  Fetches and returns the features supported by the Cluster Network Type
+ *  (Openshift SDN or Kubernetes OVN)
+ *
+ *  @async
+ *  @returns [ClusterNetworkFeatures, boolean, any] - The asynchronously-loaded cluster network
+ *  features, plus a boolean that is 'false' until the first value is loaded (or an error is
+ *  returned)
+ */
+export const useClusterNetworkFeatures = (): [ClusterNetworkFeatures, boolean] => {
+  const [features, setFeatures] = React.useState<ClusterNetworkFeatures>();
+  const [featuresLoaded, setFeaturesLoaded] = React.useState(false);
```

Comment:
I wonder if we want to distinguish the "not yet loaded" state versus "unknown" state, or we could say it's the same thing. We could argue that features are unknown until they are loaded, and hence remove this `featuresLoaded` state. What do you think?

Reply:
I'm afraid that sometimes this could cause some kind of annoying warning message that is initially shown and immediately hidden after the page is loaded.

The other way, the unsupported fields are initially hidden and immediately shown after the page is loaded. This would give a more "natural" feeling of a page being progressively loaded.

What do you think?

Reply:
Yeah, I've tried locally, it generates some blinking effect that isn't very nice indeed. Ok to keep as you did